### ISSUE TITLE
Unsafe behavior for empty validation sets

### ISSUE THREAD

**Issue by emjotde** [marian-nmt/marian-dev#310]
Stacked bugs:
1) When given an empty validation set (0 bytes) the program still tries to validate on them which results in `-nan` for cross-entropy and perplexity. We should check for 0 bytes files.
2) When last-best in *.progress.yml containes `-nan` as a value, resuming a training results in a YAML error as YAML does not know how to convert `-nan` to a float.

**Comment by snukky** [comment]
I tried:
* empty validation source and target sets for ce/bleu/translation of a combination of them
* empty validation source or target set
* empty validation set for training a language model with perplexity

and all of these report correctly "File valid.xyz is empty" with the current marian-dev.
Empty 0 bytes files were created by `touch valid.src`.

The only case I've found unintuitive is when source and target valid sets differ in sizes as there is no warning or error.

**Comment by emjotde** [comment]
I have to logs to prove it happened, but cannot reproduce this either now. Let's close the issue and reopen if it occurs again.

**Comment by snukky** [comment]
OK. Added regression tests: https://github.com/marian-nmt/marian-regression-tests/commit/020f942e53e6d6f06828a0d962621b0f7ccebf52

**Comment by snukky** [comment]
Actually there is something wrong.
